app/src/main/java/com/zgamelogic/ghosty/MainActivity.java

```diff
@@ -2,13 +2,100 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.util.Log;
+import android.view.View.OnClickListener;
 import android.os.Bundle;
 
-public class MainActivity extends AppCompatActivity {
+import com.zgamelogic.ghosty.data.Ghost;
+import com.zgamelogic.ghosty.services.GhostyAPI;
+
+import java.util.LinkedList;
+
+
+public class MainActivity<dataSet> extends AppCompatActivity {
+
+    InvestigationViewManager ivManager;
+    OnClickListener evListener, ghostListener;
+
+    final Integer SHORT_WAIT = 200;  // ms
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        LinkedList<String> evidenceListItems = getEvidenceListData();
+        LinkedList<Ghost> ghostListItems = getGhostListData();
+
+        ivManager =  new InvestigationViewManager(this, ghostListItems, evidenceListItems);
+    }
+
+    @Override
+    //Back key detection
+    public void onBackPressed()
+    {
+        super.onBackPressed();
+    }
+
+    /**
+     * Get information for evidence list data for RecyclerView
+     * This will signal to InvestigationViewManager when the list is ready.
+     * @return Arraylist<String> containing data for evidence list
+     */
+    private LinkedList<String> getEvidenceListData()
+    {
+        LinkedList<String> list = new LinkedList<>();
+        GhostyAPI.getEvidence(evidences -> {
+            for(String evidence: evidences){
+                Log.d(GhostyAPI.LOG_API, "we read: " + evidence);
+                list.add(evidence);
+            }
+
+            // Update UI once recycler view adapter support is initialized
+            while (ivManager == null) {
+                try {
+                    Thread.sleep(SHORT_WAIT);
+                } catch (InterruptedException e) {
+                    Log.e(GhostyAPI.LOG_API, "interrupted evidence api sleep");
+                }
+            }
+            runOnUiThread(new Runnable() {
```

Comment:
this can be a lambda
```suggestion
            runOnUiThread(() -> ivManager.populateEvidenceList());
```